scripts/setup_linux.sh

```diff
@@ -1,61 +1,254 @@
+LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco"
+LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/mjpro150/bin\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
+  printf "export ${LD_LIB_ENV_VAR}\\n" >> "${BASH_RC}"
 fi
 
-echo "Creating conda environment..."
+# Set up conda
+hash conda 2>/dev/null || {
+  wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh \
+    -O miniconda.sh
+  sudo bash miniconda.sh -b -p /opt/conda
+  rm miniconda.sh
+  sudo ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh
+}
+# Add conda to executable programs
+export PATH="$PATH:/opt/conda/bin"
+PATH_ENV_VAR="PATH=\"\$PATH:/opt/conda/bin\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
+  printf "export ${PATH_ENV_VAR}\\n" >> "${BASH_RC}"
+fi
+conda update -q -y conda
+
+# Create conda environment
 conda env create -f environment.yml
 if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: conda environment could not be created\e[0m"
+  printf '\e[0;31m%s\e[0m\n' "Error: conda environment could not be created"
   exit 1
 fi
+sudo rm -rf /opt/conda/pkgs/*
 
-env_name="garage"
-tf_version="1.8"
-read -r -p "Install tensorflow-gpu instead of regular tensorflow [y/N]: " \
-  response
-case "${response}" in
-  [yY])
-    source activate "${env_name}"
-    pip install tensorflow-gpu=="${tf_version}"
-    source deactivate
-    ;;
-  *)
-    source activate "${env_name}"
-    pip install tensorflow=="${tf_version}"
-    source deactivate
-    ;;
-esac
-
-echo "Updating conda environment..."
-conda env update
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: conda environment could not be updated\e[0m"
-  exit 1
+# Extras
+source activate garage
+# Prevent pip from complaining about available upgrades
+pip install --upgrade pip
+# TensorFlow is not in environment.yml because of CPU vs GPU
+# Remove any TensorFlow installations before installing the desired flavor
+pip uninstall -y tensorflow tensorflow-gpu
+if [[ "${_arg_tf_gpu}" = on ]]; then
+  pip install tensorflow-gpu
+else
+  pip install tensorflow
+fi
+# Fix Box2D install
+# See https://github.com/openai/gym/issues/100
+# See https://github.com/pybox2d/pybox2d/issues/82
+pip uninstall -y Box2D Box2D-kengz box2d-py
+pip install Box2D
+# We need a MuJoCo key to import mujoco_py
+cp ${_arg_mjkey_path} "${HOME}"/.mujoco/mjkey.txt
+python -c 'import mujoco_py'
+# Set up pre-commit in local repo
+pre-commit install -t pre-commit
+pre-commit install -t pre-push
+pre-commit install -t commit-msg
+source deactivate
+
+# Add garage to python modules
+export PYTHONPATH="$PYTHONPATH:${GARAGE_DIR}"
+PYTHON_ENV_VAR="PYTHONPATH=\"\$PYTHONPATH:${GARAGE_DIR}\""
+if [[ "${_arg_set_envvar}" = on ]]; then
+  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
+  printf "export ${PYTHON_ENV_VAR}\\n" >> "${BASH_RC}"
```

Comment:
It looks like you are bending over backwards with a lot of escapes in these. You can usually use `echo -e` to make things simpler